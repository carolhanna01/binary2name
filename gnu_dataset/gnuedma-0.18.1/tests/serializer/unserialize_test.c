/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdio.h>
#include <edma.h>

int main (int argc, char *argv[])
{
  OBJID		id, id1;
  ESint32       n, i;
  EChar         v[80],v1[80],v2[200];
  EDMAT_BUFFER buf;
  
  EDMAInit();
  id1 = edma_new_obj ("SERIALIZER");
  id = edma_new_obj ("ID");
  
  /*edma_met3 (id, "SetDB", "test.db");*/

  edma_met3 (id1, "load", "data.dat");  
  edma_met3 (id1, "unmarshall", &id);

  edma_rprop3 (id, "nElems", &n);
  edma_rprop3 (id, "List", &buf);
  printf ("Showing %d elements\n", n);
  printf ("List size : %d\n", buf.Size);
  for (i = 0; i < n; i++)
    {
      edma_met3 (id, "GetItem", i, v, v1, v2);
      printf ("Item %ld: [%s] [%s] [%s]\n", i, v, v1, v2);
    }
  edma_free_obj (id);
  edma_free_obj (id1);
  EDMAEnd();
}
