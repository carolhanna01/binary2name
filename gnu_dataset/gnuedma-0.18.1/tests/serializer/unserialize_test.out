
GNU/EDMA 0.17.1 build [Feb  6 2010][Beta][0.17.1] starting up...
EDMA Startup process... OK
(SERIALIZER)[INFO] About to read 20844 bytes from file data.dat
(SERIALIZER)Class name : ID
(SERIALIZER)Pointer : 0x80643a0 [2] properties to read
(SERIALIZER)Type = 2 Size = 4
(SERIALIZER)Type = 14 Size = 20736
Showing 5 elements
List size : 20736
Item 0: [EDMAIDF_BUILDER] [idf] [Build EDMA Interface Definition File]
Item 1: [EDMA_C_IMP_BUILDER] [c] [Build C Class's Skeleton]
Item 2: [EDMA_INES_BUILDER] [ines] [Build INES files]
Item 3: [EDMA_DOC_BUILDER] [sgml] [Build SGML EDMA Classes Documentation]
Item 4: [AUTOTOOLS_PROJECT] [sgml] [Builds a basic EDMA class project]
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
